contracts/utils/TokenVestingNoStaking.sol

```diff
@@ -0,0 +1,184 @@
+ *      Therefore, it is recommended to avoid using short time durations (less than a minute).
+ *
+ */
+contract TokenVestingNoStaking is ITokenVestingNoStaking, Shared {
+    using SafeERC20 for IERC20;
+
+    uint256 public constant CLIFF_DENOMINATOR = 5; // x / 5 = 20% of x
+
+    // beneficiary of tokens after they are released. It can be transferrable.
+    address private beneficiary;
+    bool public immutable transferableBeneficiary;
+    // the revoker who can cancel the vesting and withdraw any unvested tokens
+    address private revoker;
+
+    // Durations and timestamps are expressed in UNIX time, the same units as block.timestamp.
+    uint256 public immutable cliff;
+    uint256 public immutable end;
+
+    mapping(IERC20 => uint256) public released;
+    bool public revoked;
+
+    /**
+     * @param beneficiary_ address of the beneficiary to whom vested tokens are transferred
+     * @param revoker_   the person with the power to revoke the vesting. Address(0) means it is not revocable.
+     * @param cliff_ the unix time of the cliff, nothing withdrawable before this
+     * @param end_ the unix time of the end of the vesting period, everything withdrawable after
+     * @param transferableBeneficiary_ whether the beneficiary address can be transferred
+     */
+    constructor(
+        address beneficiary_,
+        address revoker_,
+        uint256 cliff_,
+        uint256 end_,
+        bool transferableBeneficiary_
+    ) nzAddr(beneficiary_) {
+        require(cliff_ <= end_, "Vesting: cliff_ after end_");
+        require(block.timestamp < cliff_, "Vesting: cliff before current time");
+
+        beneficiary = beneficiary_;
+        revoker = revoker_;
+        cliff = cliff_;
+        end = end_;
+        transferableBeneficiary = transferableBeneficiary_;
+    }
+
+    //////////////////////////////////////////////////////////////
+    //                                                          //
+    //                  State-changing functions                //
+    //                                                          //
+    //////////////////////////////////////////////////////////////
+
+    /**
+     * @notice Transfers vested tokens to beneficiary.
+     * @param token ERC20 token which is being vested.
+     */
+    function release(IERC20 token) external override onlyBeneficiary {
+        uint256 unreleased = _releasableAmount(token);
+        require(unreleased > 0, "Vesting: no tokens are due");
+
+        released[token] += unreleased;
+        emit TokensReleased(token, unreleased);
+
+        token.safeTransfer(beneficiary, unreleased);
+    }
+
+    /**
+     * @notice Allows the revoker to revoke the vesting. Tokens already vested remain
+     *         in the contract for the beneficiary to release, the rest are returned
+     *         to the revoker.
+     * @param token ERC20 token which is being vested.
+     */
+    function revoke(IERC20 token) external override onlyRevoker {
```

Comment:
Passing in a `IERC20 token` seems like a footgun we can avoid. We can bake this into the constructor. 

This contract doesn't support more than one token anyway since once you `revoke` all tokens are unlocked in the contract unless we had a `mapping (IERC20 => bool)` which mapped the `revoke` on a per-token basis.

It simplifies the code as well, allowing us to remove the token mapping construct.